index.js

```diff
@@ -93,13 +94,15 @@ function _javascript (entry, opts, setCss) {
     cache: {}
   }
 
-  opts = xtend(base, opts || {})
+  const jsOpts = xtend(base, opts.js || {})
 
-  const b = (this.optimize)
-    ? browserify(opts)
-    : watchify(browserify(opts))
-  b.plugin(cssExtract, { out: createCssStream })
-  b.transform(sheetify)
+  const b = opts.optimize
```

Comment:
@bcomnes `this` is not in scope, I just checked :) It should not be in scope, since `_javascript` is not nested within the constructor.